ui/pages/Project/selectors.js

```diff
@@ -55,21 +54,35 @@ export const getCurrentProject = createSelector(
 const selectEntitiesForProjectGuid = (entitiesGroupedByProjectGuid, projectGuid) => entitiesGroupedByProjectGuid[projectGuid] || {}
 export const getProjectFamiliesByGuid = createSelector(getFamiliesGroupedByProjectGuid, getProjectGuid, selectEntitiesForProjectGuid)
 export const getProjectAnalysisGroupsByGuid = createSelector(getAnalysisGroupsGroupedByProjectGuid, getProjectGuid, selectEntitiesForProjectGuid)
+const getProjectSamplesByGuid = createSelector(getSamplesGroupedByProjectGuid, getProjectGuid, selectEntitiesForProjectGuid)
+
+const getAnalysisGroupGuid = (state, props) => ((props || {}).match ? props.match.params.analysisGroupGuid : (props || {}).analysisGroupGuid)
+
+const getCurrentAnalysisGroup = createSelector(
+  getProjectAnalysisGroupsByGuid,
+  getAnalysisGroupGuid,
+  (projectAnalysisGroupsByGuid, analysisGroupGuid) => analysisGroupGuid && projectAnalysisGroupsByGuid[analysisGroupGuid],
+)
 
 export const getProjectAnalysisGroupFamiliesByGuid = createSelector(
   getProjectFamiliesByGuid,
-  getProjectAnalysisGroupsByGuid,
-  (state, props) => (props.match ? props.match.params.analysisGroupGuid : props.analysisGroupGuid),
-  (projectFamiliesByGuid, projectAnalysisGroupsByGuid, analysisGroupGuid) => {
-    if (!analysisGroupGuid || !projectAnalysisGroupsByGuid[analysisGroupGuid]) {
+  getCurrentAnalysisGroup,
+  (projectFamiliesByGuid, analysisGroup) => {
+    if (!analysisGroup) {
       return projectFamiliesByGuid
     }
-    return projectAnalysisGroupsByGuid[analysisGroupGuid].familyGuids.reduce(
+    return analysisGroup.familyGuids.reduce(
       (acc, familyGuid) => ({ ...acc, [familyGuid]: projectFamiliesByGuid[familyGuid] }), {},
     )
   },
 )
 
+export const getProjectAnalysisGroupIndividualsCount = createSelector(
+  getProjectAnalysisGroupFamiliesByGuid,
+  familiesByGuid =>
+    Object.values(familiesByGuid).reduce((acc, family) => acc + family.individualGuids.length, 0),
```

Comment:
running a count is much faster than keeping track of all the objects and then counting the lengths at the end, especially because it saves a ton of memory usage